### PR TITLE
Fixing upgrade bug

### PR DESCRIPTION
**Pull request by freddyucv**
When try to run the Upgrade task it fails because  the new field is added into a primary key and it not have any default value so for all the register that already exists the values is NULL, and then it throw a Exception when try to add the field into the primary key.

Also I did refactoring

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task221018CreateVariantFieldInMultiTree.java [hunk 1]
```diff
@@ -25,30 +28,58 @@ private boolean hasAddRemoveField() {
...
+    private boolean primaryKeyContainsVariantId() {
+        final DotDatabaseMetaData databaseMetaData = new DotDatabaseMetaData();
+        final List<String> primaryKeysFields = databaseMetaData.getPrimaryKeysFields("multi_tree");
+
+        return primaryKeysFields.contains("variant_id");
+    }
+
     public String getPostgresScript() {
-        return hasAddRemoveField() ? StringPool.BLANK : getCreateFieldStatement() + ";" + addVariantIdIntoPrimaryKey() + ";";
+        return !forceRun() ? StringPool.BLANK : getStatements();
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/startup/runonce/Task221018CreateVariantFieldInMultiTree.java:39]: double check, yeah why not

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task221018CreateVariantFieldInMultiTreeTest.java [hunk 2]
```diff
@@ -94,16 +111,18 @@ private static void cleanUp () throws SQLException {
                 .orElse("multi_tree_pkey");
 
         try {
-            new DotConnect().executeStatement(
-                    "ALTER TABLE multi_tree DROP COLUMN variant_id");
-        } catch (SQLException e) {
+            DotDatabaseMetaData.dropPrimaryKey("multi_tree");
+        } catch (Exception e) {
             //ignore
+            System.out.println();
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task221018CreateVariantFieldInMultiTreeTest.java:117]: can this be removed?

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task221018CreateVariantFieldInMultiTreeTest.java [hunk 3]
```diff
@@ -94,16 +111,18 @@ private static void cleanUp () throws SQLException {
...
         }
 
         try {
-            DotDatabaseMetaData.dropPrimaryKey("multi_tree");
-        } catch (Exception e) {
+            new DotConnect().executeStatement(
+                    "ALTER TABLE multi_tree DROP COLUMN variant_id");
+        } catch (SQLException e) {
             //ignore
+            System.out.println();
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task221018CreateVariantFieldInMultiTreeTest.java:125]: same here

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task221018CreateVariantFieldInMultiTreeTest.java (hunk 2 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task221018CreateVariantFieldInMultiTreeTest.java]: done https://github.com/dotCMS/core/pull/23270/commits/f2da06cdf6fa6d1041394dc24d1d0f0b03fa972f

File: dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task221018CreateVariantFieldInMultiTreeTest.java (hunk 3 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task221018CreateVariantFieldInMultiTreeTest.java]: done https://github.com/dotCMS/core/pull/23270/commits/f2da06cdf6fa6d1041394dc24d1d0f0b03fa972f

**Pull request merged**